hermeto/core/package_managers/cargo/main.py

```diff
@@ -146,33 +147,88 @@ def _hidden_cargo_config_file(package_dir: RootedPath) -> Generator[None, None,
         config.path.write_text(data) if data is not None else do_nothing
 
 
+@contextmanager
+def _temporary_cwd(path_to_new_cwd: Path) -> Generator[None, None, None]:
+    oldcwd = os.getcwd()
+    os.chdir(path_to_new_cwd)
+    yield
+    os.chdir(oldcwd)
+
+
+def _run_cmd_watching_out_for_lock_mismatch(
+    cmd: list, params: dict, package_dir: RootedPath, mode: Mode
+) -> str:
+    warn_about_imminent_update_to_cargo_lock = (
+        f"A mismatch between Carg.lock and Cargo.toml was detected in {package_dir}. "
+        "Because of permissive mode Hermeto will now regenerate the lock file "
+        "to match expectation and will try to process the package again. This "
+        f"is a violation of reproducibility and must be addressed by {package_dir.path.name} "
+        "maintainers."
+    )
+    update_cargo_lock_cmd = ["cargo", "generate-lockfile"]
+    # The four lines below monkey-patch logging in run_cmd for the purpose of hiding an error
+    # report when a user is willing to override it (an error would otherwise be reported
+    # unconditionally followed by a warning which among other things tells a user that steps
+    # were taken to work around the error). It is done here like this to address a single
+    # corner case which at the moment does not warrant adding complexity to run_cmd itself.
+    # This approach should be revisited once there is more need for fine-tuning its output.
+    # Unpatching happens unconditinally on leaving the local environment.
+    from hermeto.core.utils import log as ulog
+
+    old_log_error = ulog.error
+    error_accumulator = []
+    # Mypy does not believe in method assignment.
+    ulog.error = lambda *a, **k: error_accumulator.append((a, k))  # type: ignore[assignment]
+
```

Comment:
Please don't do this. It is exactly these "single corner case" handling hunks that will ultimately add to the tech debt in the future when it is no longer just about a single corner case. If you need to silence `run_cmd` then I'd suggest using a new argument `silence` for `run_cmd` with a default value of False and take care of the rest here based on the permissive mode setting.